views/window/dockedwindow.html

```diff
@@ -1,47 +1,47 @@
-          {{label}}
-        </span>
+  <div class="modal-content d-flex flex-column flex-fill" ng-class="{'bg-transparent': overlay, 'border-0': !border}">
+    <div class="c-docked-window__buttons">
+      <span class="text-truncate flex-fill" title="{{label}}">
+        <i ng-if="icon" ng-class="icon" class="mr-1"></i>
+        <span ng-if="iconHtml" ng-bind-html="iconHtml"></span>
+        {{label}}
+      </span>
 
-        <span ng-if="helpContext" class="d-inline-flex">
-          <context-onboarding context="{{helpContext}}"></context-onboarding>
-        </span>
+      <span ng-if="helpContext" class="d-inline-flex">
+        <context-onboarding context="{{helpContext}}"></context-onboarding>
+      </span>
 
-        <span ng-repeat="headerBtnCfg in headerBtns | reverse"
-            ng-click="windowCtrl.onHeaderBtnClick($event, headerBtnCfg);"
-            title="{{headerBtnCfg.tooltip}}">
-          <i class="{{'c-glyph ' + headerBtnCfg.iconClass}}"></i>
-        </span>
+      <span ng-repeat="headerBtnCfg in headerBtns | reverse"
+          ng-click="windowCtrl.onHeaderBtnClick($event, headerBtnCfg);"
+          title="{{headerBtnCfg.tooltip}}">
+        <i class="{{'c-glyph ' + headerBtnCfg.iconClass}}"></i>
+      </span>
 
-        <span ng-if="showCollapse" 
-            ng-click="windowCtrl.toggle()" 
-            title="{{collapsed ? 'Expand' : 'Collapse'}}" 
-            class="d-inline-flex">
-          <i class="fa c-glyph window-close" ng-class="{'fa-compress': !collapsed, 'fa-expand': collapsed}"></i>
-        </span>
+      <span ng-if="showCollapse"
+          ng-click="windowCtrl.toggle()"
+          title="{{collapsed ? 'Expand' : 'Collapse'}}"
+          class="d-inline-flex">
+        <i class="fa c-glyph window-close" ng-class="{'fa-compress': !collapsed, 'fa-expand': collapsed}"></i>
+      </span>
 
-        <span ng-if="showClose" 
-            ng-click="windowCtrl.close(true)" 
-            class="close flex-shrink-0" 
-            aria-label="Close">
-          <span class="c-glyph" aria-hidden="true">&times;</span>
-        </span>
+      <button ng-if="showClose"
+          class="btn border-0 p-0 close flex-shrink-0"
+          type="button"
+          ng-click="windowCtrl.close(true)"
+          aria-label="Close">
+        <span class="c-glyph" aria-hidden="true">&times;</span>
+      </button>
 
-        <span ng-if="showHide" 
-            type="button" 
-            class="close flex-shrink-0" 
-            ng-click="windowCtrl.hide()" 
-            aria-label="Hide">
-          <span class="c-glyph" aria-hidden="true">&times;</span>
-        </span>
-      </div>
-      <div class="c-docked-window__content w-100 flex-fill d-flex flex-column">
-        <div class="js-window__wrapper d-flex flex-column flex-fill" ng-transclude></div>
-      </div>
+      <button ng-if="showHide"
+          class="btn border-0 p-0 close flex-shrink-0"
+          type="button"
```

Comment:
This might have been the source. I think type=button isn't valid for spans. But swapping to actual buttons is better overall, so I'm good with that!

Reply:
Even without `type="button"`, a `span` does not meet the criteria for the rule. It would need a `role` that indicates it's interactive like `role="button"`. We could disable the rule if we disagree with its behavior, it [seems the intent](https://gitlab.com/html-validate/html-validate/-/issues/110) was to avoid overuse of the attribute but this is a legitimate use for it.